tiledb/sm/c_api/tiledb.h

```diff
@@ -2083,21 +2080,34 @@ TILEDB_EXPORT int tiledb_array_alloc(
     tiledb_ctx_t* ctx, const char* array_uri, tiledb_array_t** array);
 
 /**
- * Opens a TileDB array.
+ * Opens a TileDB array. The array is opened using a query type as input.
+ * This is to indicate that queries created for this array object will
+ * inherit the query type. In other words, array objects are opened to
+ * receive only one type of queries. They can always be closed and
+ * be re-opened with another query type. Also there may be many different
+ * array objects created and opened with different query types. For
+ * instance, one may create and open an array object `array_read` for
+ * reads and another one `array_write` for writes, and interleave
+ * creation and submission of queries for both these array objects.
```

Comment:
Need to clarify here what is meant by "array object." Is an array object the directory on disk? Or the `tiledb_array_t`?